### PR TITLE
Remove duplicated CF forms code

### PR DESCRIPTION
**Pull request by jimmynotjim**
The enhancements code for cf-forms is no longer needed now that the
project is using Capital Framework v4

## Removals

- All of the styles in the cf-forms enhancements that already exist in Capital Framework

## Testing

1. Run `gulp build`, make sure it doesn't error
2. Navigate through the site and ensure the forms still look the same.

## Todos

- Move the rest of the enhancements to Capital Framework.

## Checklist

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated
* [x] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/unprocessed/css/enhancements/forms.less [hunk 1]
```diff
@@ -341,15 +196,8 @@ select[multiple] {
     }
 }
 
-/* topdoc
-  name: Sets default styling for multi-selects
- */
 
-select[multiple] {
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/forms.less]: Hmm I can't find this style in CF, are you sure it's not needed?

File: cfgov/unprocessed/css/enhancements/forms.less [hunk 2]
```diff
@@ -376,22 +224,7 @@ input {
...
-  name: Ensure form legends wrap in IE
-  family: cf-form
-  notes:
-    - "Legends do not wrap in IE. Different styles are
-      required to ensure wrapping in different versions."
-  tags:
-    - cf-form
-*/
-
-legend {
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/forms.less]: This one too, I don't see it in CF.

File: cfgov/unprocessed/css/enhancements/forms.less [hunk 3]
```diff
@@ -1,149 +1,4 @@
...
-        margin: 0;
-        padding: 0;
-
-        .m-field + .m-field {
-            margin-top: unit( 10px / @base-font-size-px, em );
-        }
-    }
-}
-
-.o-form_fieldset .a-text-input {
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/forms.less]: `o-form_fieldset` is referenced in the CF docs but doesn't have this style, so the consequence looks like any text inputs won't be 100% when inside a fieldset, if that's in use somewhere.

Comment (anselmbradford) [inline cfgov/unprocessed/css/enhancements/forms.less]: `o-form_fieldset` exists in CF in `form.less`, but not with these styles.

Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: These are used inside "input with button" https://github.com/cfpb/capital-framework/blob/canary/src/cf-forms/src/organisms/form.less#L41-L44

**Comment by anselmbradford** [comment]
@jimmynotjim You might want to pull the latest cf-forms into this PR, there was another release last night. Thanks for your work here!

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: https://github.com/cfpb/capital-framework/blob/canary/src/cf-forms/src/atoms/multiselect.less

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 2 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: https://github.com/cfpb/capital-framework/blob/canary/src/cf-forms/src/atoms/legend.less

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 3 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: This is duplicated in cfgov here https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/unprocessed/css/organisms/form.less

But I also have a PR to remove it entirely here https://github.com/cfpb/cfgov-refresh/pull/3386

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: Ah. Okay if the markup is updated 👍

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 2 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: Okay if markup is updated 👍

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: Oh ok, I thought that happened during the v4 work. I'll search it out for this and the one below

**Comment by cfarm** [comment]
Yay, this is great! Thanks Jimmy!

**Comment by jimmynotjim** [comment]
@anselmbradford should we update CF deps in their own PRs? Seems out of scope, especially if there's any bugs or changes required here.

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: Hopefully is, but good time to check when the CSS is removed.

**Comment by anselmbradford** [comment]
> should we update CF deps in their own PRs? Seems out of scope, especially if there's any bugs or changes required here.

Sure if that works for you. I just figured since `m-form-field` is updated in the latest CF and you're looking at it here you may want to be looking at the latest that CF is bringing in.

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: We don't seem to be using multiselects outside of those that are controlled by the Multiselect scripting so I think this is safe. That said, I still discovered a bug [I've logged here](https://github.com/cfpb/cfgov-refresh/issues/3443)

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 2 above)
Comment (jimmynotjim) [reply cfgov/unprocessed/css/enhancements/forms.less]: The only one I'm not sure about [is this](https://github.com/cfpb/cfgov-refresh/blob/9964c5cb3a4bfcee228f0184a1a0032f92f343b9/cfgov/jinja2/v1/_includes/conference-registration-form.html#L90). Is this still in use?

Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: As far as I know yes, annually. There's an option in Wagtail to add the conference registration form. @chosak knows.

File: cfgov/unprocessed/css/enhancements/forms.less (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/css/enhancements/forms.less]: 👍

**Comment by anselmbradford** [comment]
Something is up with the blog filter that isn't happening on the live site (resize window down):

screen shot 2017-10-06 at 11 21 41 am

**Comment by jimmynotjim** [comment]
@anselmbradford good catch. I forgot that all input + label combos need to be wrapped in a field to get the proper layout styles. Should be fixed.

**Comment by anselmbradford** [comment]
Looks good now 👍

**Pull request merged**